### PR TITLE
Not all qubits are in the initial map

### PR DESCRIPTION
**Pull request by yao-cqc**
@cqc-alec @cqc-melf this branch contains a test that produces the issue. (i.e. q[9] not in the initial map)

### CODE REVIEW

**Comment by cqc-melf** [comment]
@yao, thank you. I will take a look

**Comment by cqc-melf** [comment]
@yao-cqc I have tried to find a working case for the same circuit with the map by inserting a placement, but that is not working, too.

**Review by cqc-melf — APPROVED** [review]

**Review by sjdilkes — APPROVED** [review]
Looks good to me.

File: tket/src/Mapping/include/Mapping/MappingFrontier.hpp [hunk 1]
```diff
@@ -190,6 +190,19 @@ struct MappingFrontier {
   bool valid_boundary_operation(
       const ArchitecturePtr& architecture, const Op_ptr& op,
       const std::vector<Node>& uids) const;
+
+  /**
+   * Update a qubit mapping in both the initial map and the final map
+   * @param qubit the qubit mapping to be updated
+   * @param node the new node to be mapped
+   */
+  void update_bimaps(UnitID qubit, UnitID node);
```
[the remark this answers is not in the document]
Comment (yao-cqc) [reply tket/src/Mapping/include/Mapping/MappingFrontier.hpp:199]: My intension is that the routing methods are not encouraged to interact with the maps directly (although ``this->bimaps_`` remains public).

**Review by cqc-alec — APPROVED** [review]

File: tket/src/Mapping/include/Mapping/MappingFrontier.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (cqc-alec) [reply tket/src/Mapping/include/Mapping/MappingFrontier.hpp:199]: Thanks, all looks good to me.

**Pull request merged**